m4/pdns_check_libcrypto.m4

```diff
@@ -32,31 +29,31 @@
 #   and this notice are preserved. This file is offered as-is, without any
 #   warranty.
 
-#serial 8 (PowerDNS modified)
+#serial 1
 
-AU_ALIAS([CHECK_SSL], [AX_CHECK_OPENSSL])
-AC_DEFUN([AX_CHECK_OPENSSL], [
+AU_ALIAS([CHECK_LIBCRYPTO], [PDNS_CHECK_LIBCRYPTO])
+AC_DEFUN([PDNS_CHECK_LIBCRYPTO], [
     found=false
-    AC_ARG_WITH([openssl],
-        [AS_HELP_STRING([--with-openssl=DIR],
+    AC_ARG_WITH([libcrypto],
+        [AS_HELP_STRING([--with-libcrypto=DIR],
             [root of the OpenSSL directory])],
         [
             case "$withval" in
             "" | y | ye | yes | n | no)
-            AC_MSG_ERROR([Invalid --with-openssl value])
+            AC_MSG_ERROR([Invalid --with-libcrypto value])
               ;;
             *) ssldirs="$withval"
               ;;
             esac
         ], [
             # if pkg-config is installed and openssl has installed a .pc file,
             # then use that information and don't search ssldirs
-            AC_PATH_PROG([PKG_CONFIG], [pkg-config])
+            AC_CHECK_TOOL([PKG_CONFIG], [pkg-config])
             if test x"$PKG_CONFIG" != x""; then
-                OPENSSL_LDFLAGS=`$PKG_CONFIG libcrypto --libs-only-L 2>/dev/null`
+                LIBCRYPTO_LDFLAGS=`$PKG_CONFIG openssl --libs-only-L 2>/dev/null`
                 if test $? = 0; then
-                    OPENSSL_LIBS=`$PKG_CONFIG libcrypto --libs-only-l 2>/dev/null`
-                    OPENSSL_INCLUDES=`$PKG_CONFIG libcrypto --cflags-only-I 2>/dev/null`
+                    LIBCRYPTO_LIBS='-lcrypto' # This *will* break if the linker name is not crypto
+                    LIBCRYPTO_INCLUDES=`$PKG_CONFIG openssl --cflags-only-I 2>/dev/null`
```

Comment:
Why are you reverting to openssl.pc here (and above)? Is there no useful libcrypto.pc in openssl 1.1.0?